File: Python/Stack.py

class Node(object):
    def __init__(self, data, next=None):
        self.data = data
        self.next = next

class Stack(object):

    def __init__(self, head=None):
        self.head = head

    def push(self, data):
        self.head = Node(data, self.head)

    def pop(self):
        if self.is_empty():
            raise Exception("Stack is empty.")
        else:
            data = self.head.data
            self.head = self.head.next
            return data

    def peek(self):
        if self.is_empty():
            raise Exception("Stack is empty.")
        else:
            return self.head.data

    def is_empty(self):
        return self.head is None


class MyQueue(object):
    def __init__(self):
        self.enqueue_order = Stack()
        self.dequeue_order = Stack()
        self.is_in_enqueued_order = True

    # def enqueue(self, data):
    #     if not self

File: Java/stack.java
class stack{
    private int maxSize;
    private long[] stackArray;
    private int top;

    // constructor
    public stack(int s){
        maxSize = s;
        stackArray = new long[maxSize];
        top = -1;
    }

    public void push(long j){
        stackArray[++top] = j;
    }

    public long pop() {
        return stackArray[top--];
    }

    public long peek(){
        return stackArray[top];
    }
    public boolean isEmpty(){
        return (top == -1);
    }

    public boolean isFull(){
        return(top == maxSize - 1);
    }

    public static void main(String[] args){
        stack theStack = new stack(10);

        theStack.push(20);
        theStack.push(40);
        theStack.push(60);
        theStack.peek();
        theStack.push(80);
        System.out.println("Here is the stack: ");
        while(!theStack.isEmpty()){
            long value = theStack.pop();
            System.out.println(value);
        }
    } 
}